File: src/types/domain.ts
export type Language = "zh" | "en";
export type Currency = "CNY" | "USD";

export type Profile = {
  id: string;
  displayName: string;
  avatarUrl: string;
  preferredLanguage: Language;
  preferredCurrency: Currency;
};

export type FirePlan = {
  id: string;
  userId: string;
  annualSpending: number;
  withdrawalRate: number;
  targetAmount: number;
  currency: Currency;
  updatedAt: string;
};

export type Checkin = {
  id: string;
  planId: string;
  checkinDate: string;
  currentNetWorth: number;
  annualSpending: number;
  note: string;
  createdAt: string;
};

export type AppSettings = {
  language: Language;
  currency: Currency;
  defaultWithdrawalRate: number;
  demoMode: boolean;
};

File: FireCompanion/FireCompanion/Models/Domain.swift
import Foundation

typealias Language = String
extension Language {
    static let zh: Language = "zh"
    static let en: Language = "en"
}

typealias Currency = String
extension Currency {
    static let cny: Currency = "CNY"
    static let usd: Currency = "USD"
}

struct Profile: Codable, Identifiable {
    let id: String
    var displayName: String
    var avatarUrl: String
    var preferredLanguage: Language
    var preferredCurrency: Currency
}

struct FirePlan: Codable, Identifiable {
    let id: String
    let userId: String
    var annualSpending: Double
    var withdrawalRate: Double
    var targetAmount: Double
    var currency: Currency
    var updatedAt: Date
}

struct Checkin: Codable, Identifiable {
    let id: String
    let planId: String
    var checkinDate: Date
    var currentNetWorth: Double
    var annualSpending: Double
    var note: String
    var createdAt: Date
}

struct AppSettings {
    var language: Language = .zh
    var currency: Currency = .cny
    var defaultWithdrawalRate: Double = 0.04
    var demoMode: Bool = true
}
